core/modules/simpletest/simpletest.module

```diff
@@ -539,6 +576,45 @@ function simpletest_clean_results_table($test_id = NULL) {
   return 0;
 }
 
+
+/**
+ * Removes cached profile tables from the database.
+ */
+function simpletest_clean_profile_cache_tables(){
+  $tables = db_find_tables(Database::getConnection()->prefixTables('{simpletest_cache_}') . '%');
+  $count = 0;
+  foreach ($tables as $table) {
+    db_drop_table($table);
+    $count++;
+  }
+
+  if ($count > 0) {
+    backdrop_set_message(format_plural($count, 'Removed 1 profile cache table.', 'Removed @count profile cache tables.'));
+  }
+  else {
+    backdrop_set_message(t('No profile cache tables to remove.'));
+  }
+}
+
+/**
+ * Removes cached profile folders from the files directory.
+ */
+function simpletest_clean_profile_cache_folders(){
+  $profiles = array(
```

Comment:
We are using the same list of profiles earlier, in `simpletest_run_tests()`. Should we define those in a `simpletest_get_profiles()` function, or a constant?

Reply:
I think @klonos's suggestion is good here. It would also be a good idea to make it so that new profiles or custom/contrib profiles worked with profile caching.

We already have a function in install.core.inc that finds all profiles, `install_find_profiles()`. Unfortunately this isn't a commonly included file. I think making a wrapper function like `simpletest_get_profiles()` that wraps around that other function and include would be appropriate to make it dynamic and avoid a hard-coded list.